XDCx/token.go

```diff
@@ -0,0 +1,78 @@
+package XDCx
+
+import (
+	"github.com/XinFinOrg/XDPoSChain/contracts/XDCx/contract"
+	"github.com/XinFinOrg/XDPoSChain/log"
+	"math/big"
+	"strings"
+
+	"github.com/XinFinOrg/XDPoSChain"
+	"github.com/XinFinOrg/XDPoSChain/accounts/abi"
+	"github.com/XinFinOrg/XDPoSChain/accounts/abi/bind/backends"
+	"github.com/XinFinOrg/XDPoSChain/common"
+	"github.com/XinFinOrg/XDPoSChain/consensus"
+	"github.com/XinFinOrg/XDPoSChain/core/state"
+)
+
+// GetTokenAbi return token abi
+func GetTokenAbi() (*abi.ABI, error) {
+	contractABI, err := abi.JSON(strings.NewReader(contract.TRC21ABI))
+	if err != nil {
+		return nil, err
+	}
+	return &contractABI, nil
+}
+
+// RunContract run smart contract
+func RunContract(chain consensus.ChainContext, statedb *state.StateDB, contractAddr common.Address, abi *abi.ABI, method string, args ...interface{}) (interface{}, error) {
+	input, err := abi.Pack(method)
+	if err != nil {
+		return nil, err
+	}
+	backend := (*backends.SimulatedBackend)(nil)
+	fakeCaller := common.HexToAddress("0x0000000000000000000000000000000000000001")
+	msg := XDPoSChain.CallMsg{To: &contractAddr, Data: input, From: fakeCaller}
+	result, err := backend.CallContractWithState(msg, chain, statedb)
```

Comment:
the simulated.go is used for testing purpose. Why added functions that will be triggered during production run? for example: CallContractWithState etc(all the newly added ones)